### PR TITLE
{Azure-firewall} Migrate Azure-firewall to AAZ

### PR DESCRIPTION
**Pull request by Jing-song**
---
AAZ: https://github.com/Azure/aaz/pull/256

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

`az network firewall`

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Azure-firewall

File: src/azure-firewall/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,10 @@
 Release History
 ===============
 
+1.0.0
+++++++
```
Comment (jsntcy) [inline src/azure-firewall/HISTORY.rst:7]: Release notes should contain changes that impact customers, not internal changes.

File: src/azure-firewall/azext_firewall/azext_metadata.json [hunk 2]
```diff
@@ -1,4 +1,4 @@
 {
     "azext.isPreview": true,
```
Comment (jsntcy) [inline src/azure-firewall/azext_firewall/azext_metadata.json:2]:
```suggestion
```

File: src/azure-firewall/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,10 @@
...
+* `az network firewall`: Migrate `threat-intel-allowlis` `policy intrusion-detection` `policy rule-collection-group` to AAZ command
```
Comment (jsntcy) [inline src/azure-firewall/HISTORY.rst:8]:
```suggestion
* [Fix] `az network firewall threat-intel-allowlist create/update/delete`:  Remove the argument `--ids` as sub command should not support it.
* Make extension from preview to stable.
```

**Review by jsntcy — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Pull request merged**